enable streaming of integration test output (#21912)
pantsbuild/pants: docs/notes/2.25.x.md

The oldest [glibc version](https://www.sourceware.org/glibc/wiki/Glibc%20Timeline) supported by the published Pants wheels is now 2.28.  This should have no effect unless you are running on extremely old Linux distributions.  See <https://github.com/pypa/manylinux> for background context on Python wheels and C libraries.

> The integration testing framework in the `pantsbuild.pants.testutil` package now supports streaming the output of the Pants invocation under test to the console. This is useful when debugging long-running integration tests which would otherwise show no output while they run since the integration test framework previously only captured output to a buffer. To use, set the `stream_output` parameter to `True` on specific invocation(s) of the `run_pants_with_workdir` and `run_pants` function, and then run the test with `--debug` so the test is invoked as an interactive process and the pytest `--capture=no` option so pytest does not capture the output during the run.

Reviewer: Maybe this could be slightly more prescriptive:

```suggestion
The integration testing framework in the `pantsbuild.pants.testutil` package now supports streaming the output of the Pants invocation under test to the console. This is useful when debugging long-running integration tests which would otherwise show no output while they run since the integration test framework previously only captured output to a buffer. To use, adjust specific test(s) to use the new `stream_output` parameter, like `run_pants_with_workdir(..., stream_output=True)` or `run_pants(..., stream_output=True)`, and then run the test `pants test --debug path/to:test -- --capture=no` so the test is invoked as an interactive process and pytest does not capture the output during the run.
```

Author: > Just confirming: this is designed to be used in an adhoc way? For instance: someone is working on a specific feature with a particular integration test of interest, so they temporarily turn stream_output=True on while iterating.

Correct. This feature is really for Pants developers. For example, it could be used to understand why an integration test is hanging, where otherwise the developer would just see no output.

Author: Added to the permanent docs as suggested.
